src/parquet/column/test-util.h

```diff
@@ -177,39 +203,196 @@ void DataPageBuilder<Type::BOOLEAN>::AppendValues(const ColumnDescriptor *d,
+      encoding_(Encoding::RLE_DICTIONARY),
+      have_values_(false) {
+        int type_length = 0;
+        if (TN == Type::FIXED_LEN_BYTE_ARRAY) {
+          type_length = d->type_length();
+        }
+        encoder_.reset(new DictEncoder<TC>(&pool_, type_length));
+        encoding_ = encoding;
+  }
+
+  void AppendValues(const vector<TC>& values) {
+    int num_values = values.size();
+    // Dictionary encoding
+    for (int i = 0; i < num_values; ++i) {
+      encoder_->Put(values[i]);
+    }
+    num_dict_values_ = encoder_->num_entries();
+    have_values_ = true;
+  }
+
+  void WriteDict() {
+    shared_ptr<OwnedMutableBuffer> dict_buffer = std::make_shared<OwnedMutableBuffer>();
+    dict_buffer->Resize(encoder_->dict_encoded_size());
+    encoder_->WriteDict(dict_buffer->mutable_data());
+    sink_->Write(reinterpret_cast<const uint8_t*>(dict_buffer->data()),
+        dict_buffer->size());
+  }
+
+  int32_t num_values() const {
+    return num_dict_values_;
+  }
+
+  Encoding::type encoding() const {
+    return encoding_;
+  }
+
+  void rle_indices(shared_ptr<OwnedMutableBuffer>& rle_indices) {
+    rle_indices->Resize(sizeof(int) * encoder_->EstimatedDataEncodedSize());
+    int actual_bytes = encoder_->WriteIndices(rle_indices->mutable_data(),
+      rle_indices->size());
+    rle_indices->Resize(actual_bytes);
+  }
+
+  void free_pool() {
+    pool_.FreeAll();
+  }
+
+  void clear_indices() {
+    encoder_->ClearIndices();
+  }
+
+ private:
+  InMemoryOutputStream* sink_;
+  MemPool pool_;
+  shared_ptr<DictEncoder<TC> > encoder_;
+  int32_t num_dict_values_;
+  Encoding::type encoding_;
+
+  bool have_values_;
+};
+
+template<>
+DictionaryPageBuilder<BooleanType>::DictionaryPageBuilder(const ColumnDescriptor *d,
+    InMemoryOutputStream* sink, Encoding::type encoding) {
+  ParquetException::NYI("only plain encoding currently implemented for boolean");
+}
+
+template<>
+void DictionaryPageBuilder<BooleanType>::WriteDict() {
+  ParquetException::NYI("only plain encoding currently implemented for boolean");
+}
+
+template <typename Type>
+static shared_ptr<DictionaryPage> MakeDictPage(const ColumnDescriptor *d,
+    const vector<typename Type::c_type>& values, const vector<int>& values_per_page,
+    Encoding::type encoding, vector<shared_ptr<Buffer> >& rle_indices) {
+  InMemoryOutputStream page_stream;
+  test::DictionaryPageBuilder<Type> page_builder(d, &page_stream, encoding);
+  int num_pages = values_per_page.size();
+  int value_start = 0;
+
+  for (int i = 0; i < num_pages; i++) {
+    page_builder.AppendValues(slice(values, value_start,
+          value_start + values_per_page[i]));
+    shared_ptr<OwnedMutableBuffer> indices_buff = std::make_shared<OwnedMutableBuffer>();
+    page_builder.rle_indices(indices_buff);
+    rle_indices.push_back(indices_buff);
+    value_start += values_per_page[i];
+    page_builder.clear_indices();
+  }
+
+  page_builder.WriteDict();
+
+  auto buffer = page_stream.GetBuffer();
```

Comment:
You could consolidate some steps by having `WriteDict` return a Buffer directly instead of Buffer -> OutputStream -> Buffer, then you don't need to pass an outputstream into the ctor at all